tests/PHPStan/Rules/Comparison/ImpossibleCheckTypeFunctionCallRuleTest.php

```diff
@@ -910,7 +920,105 @@ public function testLooseComparisonAgainstEnums(): void
 				'Call to function in_array() with null and array{LooseComparisonAgainstEnums\FooBackedEnum} will always evaluate to false.',
 				96,
 			],
-		]);
+			[
+				'Call to function in_array() with LooseComparisonAgainstEnums\FooUnitEnum and array<string> will always evaluate to false.',
+				125,
+				$tipText,
+			],
+			[
+				'Call to function in_array() with arguments LooseComparisonAgainstEnums\FooUnitEnum, array<string> and false will always evaluate to false.',
+				128,
+				$tipText,
+			],
+			[
+				'Call to function in_array() with arguments LooseComparisonAgainstEnums\FooUnitEnum, array<string> and true will always evaluate to false.',
+				131,
+				$tipText,
+			],
+			[
+				'Call to function in_array() with string and array<LooseComparisonAgainstEnums\FooUnitEnum> will always evaluate to false.',
+				143,
+				$tipText,
+			],
+			[
+				'Call to function in_array() with arguments string, array<LooseComparisonAgainstEnums\FooUnitEnum> and false will always evaluate to false.',
+				146,
+				$tipText,
+			],
+			[
+				'Call to function in_array() with arguments string, array<LooseComparisonAgainstEnums\FooUnitEnum> and true will always evaluate to false.',
+				149,
+				$tipText,
+			],
+			[
+				'Call to function in_array() with LooseComparisonAgainstEnums\FooUnitEnum::B and non-empty-array<LooseComparisonAgainstEnums\FooUnitEnum::A> will always evaluate to false.',
+				159,
+				$tipText,
+			],
+			[
+				'Call to function in_array() with LooseComparisonAgainstEnums\FooUnitEnum::A and non-empty-array<LooseComparisonAgainstEnums\FooUnitEnum::A> will always evaluate to true.',
+				162,
+				$tipText,
+			],
+			[
+				'Call to function in_array() with arguments LooseComparisonAgainstEnums\FooUnitEnum::A, non-empty-array<LooseComparisonAgainstEnums\FooUnitEnum::A> and false will always evaluate to true.',
+				165,
+				'BUG',
+				//$tipText,
```

Comment:
Yeah, this a problem because TypeSpecifier is run only once on `expressionTypes`, not `nativeExpressionTypes`.

This should fix the problem (you can verify that) https://github.com/phpstan/phpstan-src/pull/2179 but comes with performance cost.